src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java

```diff
@@ -58,4 +61,81 @@ public void testGetDatabaseFromMatchesDatabaseWithEntriesWithCorrectMatch() {
         List<BibEntry> matches = new DatabaseSearcher(new SearchQuery("harrer", true, true), database).getMatches();
         assertEquals(Collections.singletonList(entry), matches);
     }
+
+    @Test
+    public void testGetDatabasefromMatchesEmptyDatabase() {
+        BibDatabase database = new BibDatabase();
+        SearchQuery query = new SearchQuery("asdf", true, true);
+
+        DatabaseSearcher databaseSearcher = new DatabaseSearcher(query, database);
+
+        assertTrue(databaseSearcher.getMatches().isEmpty());
+    }
+
+    @Test
+    public void testGetDatabasefromMatchesEmptyDatabaseWithInvalidQuery() {
+        BibDatabase database = new BibDatabase();
+        SearchQuery query = new SearchQuery("asdf[", true, true);
+
+        DatabaseSearcher databaseSearcher = new DatabaseSearcher(query, database);
+
+        assertTrue(databaseSearcher.getMatches().isEmpty());
+    }
+
+    @Test
+    public void testGetDatabasefromMatches() {
+        BibDatabase database = new BibDatabase();
+        BibEntry entry = new BibEntry();
+        entry.setType(BibtexEntryTypes.INCOLLECTION);
+        entry.setField("author", "tonho");
+        database.insertEntry(entry);
+
+        SearchQuery query = new SearchQuery("tonho", true, true);
+        DatabaseSearcher databaseSearcher = new DatabaseSearcher(query, database);
+
+        assertFalse(databaseSearcher.getMatches().isEmpty());
+    }
+
+    @Test
+    public void testGetDataBaseFromMatchesEmptyEntry() {
```

Comment:
please rename the tests...the names are rather non-expressive. One should know from the name of a failing test what went wrong. see also naming convention in the wiki https://github.com/JabRef/jabref/wiki/Code-Howtos#test-cases